clients/python/client/osparc/_utils.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Callable, Iterator, TypeVar, Union
+
+from osparc_client import (
+    File,
+    Job,
+    LimitOffsetPageFile,
+    LimitOffsetPageJob,
+    LimitOffsetPageSolver,
+    LimitOffsetPageStudy,
+    Solver,
+    Study,
+)
+
+Page = Union[
+    LimitOffsetPageJob, LimitOffsetPageFile, LimitOffsetPageSolver, LimitOffsetPageStudy
+]
+T = TypeVar("T", Job, File, Solver, Study)
+
+
+def _pagination_to_iterator(
+    pagination_method: Callable[[int, int], Page], limit: int = 20, offset: int = 0
+) -> Iterator[T]:
```

Comment:
Question, is it equivalent to return Iterator vs Generator?

Reply:
Hmm, good question. My understanding is that every generator is an iterator but not vice versa. So if that is true, then I guess this typehint is correct. But probably I should be more specific and use `typing.Generator` instead (since that is what I am actually returning). I will correct.